exchanges/kraken/kraken_test.go

```diff
@@ -1744,70 +1745,70 @@ func TestWsOrdrbook(t *testing.T) {
-        "ordertype": "limit",
-        "pair": "XBT/USD",
-        "postxid": "OGTT3Y-C6I3P-XRI6HX",
-        "price": "100000.00000",
-        "time": "1560516023.070651",
-        "type": "sell",
-        "vol": "1000000000.00000000"
-      }
-    },
-    {
-      "TDLH43-DVQXD-2KHVYY": {
-        "cost": "1000000.00000",
-        "fee": "600.00000",
-        "margin": "0.00000",
-        "ordertxid": "TDLH43-DVQXD-2KHVYY",
-        "ordertype": "limit",
-        "pair": "XBT/USD",
-        "postxid": "OGTT3Y-C6I3P-XRI6HX",
-        "price": "100000.00000",
-        "time": "1560516023.070658",
-        "type": "buy",
-        "vol": "1000000000.00000000"
-      }
-    },
-    {
-      "TDLH43-DVQXD-2KHVYY": {
-        "cost": "1000000.00000",
-        "fee": "1600.00000",
-        "margin": "0.00000",
-        "ordertxid": "TDLH43-DVQXD-2KHVYY",
-        "ordertype": "limit",
-        "pair": "XBT/USD",
-        "postxid": "OGTT3Y-C6I3P-XRI6HX",
-        "price": "100000.00000",
-        "time": "1560520332.914657",
-        "type": "sell",
-        "vol": "1000000000.00000000"
-      }
-    },
-    {
-      "TDLH43-DVQXD-2KHVYY": {
-        "cost": "1000000.00000",
-        "fee": "600.00000",
-        "margin": "0.00000",
-        "ordertxid": "TDLH43-DVQXD-2KHVYY",
-        "ordertype": "limit",
-        "pair": "XBT/USD",
-        "postxid": "OGTT3Y-C6I3P-XRI6HX",
-        "price": "100000.00000",
-        "time": "1560520332.914664",
-        "type": "buy",
-        "vol": "1000000000.00000000"
-      }
-    }
-  ],
-  "ownTrades"
-]`)
+	  [
+		{
+		  "TDLH43-DVQXD-2KHVYY": {
+			"cost": "1000000.00000",
+			"fee": "1600.00000",
+			"margin": "0.00000",
+			"ordertxid": "TDLH43-DVQXD-2KHVYY",
+			"ordertype": "limit",
+			"pair": "XBT/USD",
+			"postxid": "OGTT3Y-C6I3P-XRI6HX",
+			"price": "100000.00000",
+			"time": "1560516023.070651",
+			"type": "sell",
+			"vol": "1000000000.00000000"
+		  }
+		},
+		{
+		  "TDLH43-DVQXD-2KHVYY": {
+			"cost": "1000000.00000",
+			"fee": "600.00000",
+			"margin": "0.00000",
+			"ordertxid": "TDLH43-DVQXD-2KHVYY",
+			"ordertype": "limit",
+			"pair": "XBT/USD",
+			"postxid": "OGTT3Y-C6I3P-XRI6HX",
+			"price": "100000.00000",
+			"time": "1560516023.070658",
+			"type": "buy",
+			"vol": "1000000000.00000000"
+		  }
+		},
+		{
+		  "TDLH43-DVQXD-2KHVYY": {
+			"cost": "1000000.00000",
+			"fee": "1600.00000",
+			"margin": "0.00000",
+			"ordertxid": "TDLH43-DVQXD-2KHVYY",
+			"ordertype": "limit",
+			"pair": "XBT/USD",
+			"postxid": "OGTT3Y-C6I3P-XRI6HX",
+			"price": "100000.00000",
+			"time": "1560520332.914657",
+			"type": "sell",
+			"vol": "1000000000.00000000"
+		  }
+		},
+		{
+		  "TDLH43-DVQXD-2KHVYY": {
+			"cost": "1000000.00000",
+			"fee": "600.00000",
+			"margin": "0.00000",
+			"ordertxid": "TDLH43-DVQXD-2KHVYY",
+			"ordertype": "limit",
+			"pair": "XBT/USD",
+			"postxid": "OGTT3Y-C6I3P-XRI6HX",
+			"price": "100000.00000",
+			"time": "1560520332.914664",
```

Comment:
I considered this issue and raised it in slack but stepped away from it.
I'm already too many yaks deep on this.

For your solution above, what result would it give for the input `1590633982.571400`
As far as I can tell, it'll need refining further.

I felt as though we'd want to change `TimeFromUnixTimestampDecimal` to parse as you suggest, but then right pad it.
There's probably a better solution.
For now, This was enough for one fix, since I didn't introduce the problem 👍